lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -254,6 +299,33 @@ unsafeMkTxIn hash ix = Ledger.mkTxInPartial
 
 type TxOut era = Core.TxOut era
 
+modifyTxOutValue
+    :: RecentEra era
+    -> (Value (ShelleyLedgerEra era) -> Value (ShelleyLedgerEra era))
+    -> TxOut (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+modifyTxOutValue RecentEraBabbage f (Babbage.TxOut addr val dat script) =
+    withStandardCryptoConstraint RecentEraBabbage $
+        Babbage.TxOut addr (f val) dat script
+modifyTxOutValue RecentEraAlonzo f (Alonzo.TxOut addr val dat) =
+    withStandardCryptoConstraint RecentEraAlonzo $
+        Alonzo.TxOut addr (f val) dat
+
+modifyTxOutCoin
+    :: RecentEra era
+    -> (Coin -> Coin)
+    -> TxOut (ShelleyLedgerEra era)
+    -> TxOut (ShelleyLedgerEra era)
+modifyTxOutCoin era f = withStandardCryptoConstraint era $
+    modifyTxOutValue era (modifyCoin f)
+
+value
+    :: RecentEra era
+    -> TxOut (ShelleyLedgerEra era)
+    -> Value (ShelleyLedgerEra era)
+value RecentEraBabbage (Babbage.TxOut _ val _ _) = val
+value RecentEraAlonzo (Alonzo.TxOut _ val _) = val
```

Comment:
This function only operates on values of type `TxOut`, but the name indicates a greater level of generality.

Perhaps rename it to reflect the specificity?

```suggestion
txOutValue
    :: RecentEra era
    -> TxOut (ShelleyLedgerEra era)
    -> Value (ShelleyLedgerEra era)
txOutValue RecentEraBabbage (Babbage.TxOut _ val _ _) = val
txOutValue RecentEraAlonzo (Alonzo.TxOut _ val _) = val
```